doc/array.md

```diff
@@ -0,0 +1,237 @@
+# `array`
```

Comment:
Is there a reason for just having `array` but not the other stdlib modules? I guess this is a test or something.